test/runner/timeout_test.dart

```diff
@@ -59,6 +59,36 @@ void main() {
     test.shouldExit(1);
   });
 
+  test("timeout is reset with each retry", () {
+    d
+        .file(
+            "test.dart",
+            '''
+import 'dart:async';
+
+import 'package:test/test.dart';
+
+void main() {
+  var runCount = 0;
+  test("timeout", () async {
+    runCount++;
+    if (runCount <=2) {
+      await new Future.delayed(new Duration(milliseconds: 1000));
+    }
+  }, retry: 3);
+}
+''')
+        .create();
+
+    var test = runTest(["--timeout=0.016x", "test.dart"]);
```

Comment:
It's probably cleaner to set a time-based timeout here, rather than a factor-based one.